node/pkg/terra/watcher.go

```diff
@@ -156,6 +167,63 @@ func (e *Watcher) Run(ctx context.Context) error {
 		}
 	}()
 
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case r := <-e.obsvReqC:
+				if vaa.ChainID(r.ChainId) != vaa.ChainIDTerra {
+					panic("invalid chain ID")
+				}
+
+				tx := eth_common.BytesToHash(r.TxHash)
+
+				logger.Info("received observation request for terra",
+					zap.String("tx_hash", tx.Hex()))
+
+				client := &http.Client{
+					Timeout: time.Second * 5,
+				}
+
+				// Query for tx by hash
+				resp, err := client.Get(fmt.Sprintf("%s/cosmos/tx/v1beta1/txs/%s", e.urlLCD, strings.TrimPrefix(tx.Hex(),"0x")))
```

Comment:
In a similar vein, what's the `strings.TrimPrefix` for? Because the Eth library prepends a 0x?

Reply:
Done and tested, thanks!